scipy/sparse/linalg/dsolve/linsolve.py

```diff
@@ -177,7 +177,13 @@ def spsolve(A, b, permc_spec=None, use_umfpack=True):
             raise ValueError("convert matrix data to double, please, using"
                   " .astype(), or set linsolve.useUmfpack = False")
 
-        umf = umfpack.UmfpackContext(_get_umf_family(A))
+        umf_family = _get_umf_family(A)
+        if np.prod(A.shape) > np.iinfo(np.int32).max:
+            umf_family = umf_family[0] + "l"
+            A.indices = A.indices.astype(np.int64)
+            A.indptr = A.indptr.astype(np.int64)
```

Comment:
Should not modify the original matrix:
```
A_new = A.__class__(A.shape)
A_new.indptr = np.array(A.indptr, copy=False, dtype=np.int64)
A_new.indices = np.array(A.indices, copy=False, dtype=np.int64)
A = A_new
```
Also, astype makes always a copy, so better use `np.array` instead.

Reply:
> Should not modify the original matrix:

<facepalm>

> Also, astype makes always a copy

Good to know